### PR TITLE
netdata/packaging: Adjust auto-updater installation logic

### LINKED ISSUE #5673: netdata-updater should remain after all installations
netdata-updater.sh no longer remains in a netdata installation, unless auto-update is enabled. This is wrong, people should still be able to run the updater, even if they don't want it in cron. 
Just as with the uninstaller, the script should have never been removed. 

See where the proper place to leave the uninstaller is and put the updater there as well. (either under bin or under libexec/netdata, not sure).

The documentation should also be updated.

### PR DESCRIPTION
**Pull request by paulkatsoulakis**
##### Summary
1) Always install updater script under libexec, together with the uninstall script
2) When the user asks for auto-update, then just link the script to the cron directory
3) When the user hasnt asked for auto-update, make sure the cron link is removed

Added extra info along all those lines.
There was also an algorithmic cleanup, in the way we do the checks so things are clearer

Note: When user hasnt provided auto-update flag, means he doesnt want auto-update so we remove it.
The automatic run of the updater knows it has to use the flag (its saved in .environment), so we wont face problems there

##### Component Name
netdata/packaging/installer

##### Additional Information
Fixes #5673

### CODE REVIEW

File: netdata-installer.sh [hunk 1]
```diff
@@ -958,39 +958,58 @@ To start netdata run:
...
-		if [ -z "${crondir}" ]; then
-			echo >&2 "Cannot figure out the cron directory to install netdata-updater"
-		else
-			if [ -f "${crondir}/netdata-updater.sh" ]; then
-				progress "Removing incorrect netdata-updater filename in cron"
-				rm -f "${crondir}/netdata-updater.sh"
-			fi
-			progress "Installing new netdata-updater in cron"
+if [ -f "${INSTALLER_DIR}/packaging/installer/netdata-updater.sh" ]; then
+	sed "s|THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT|${NETDATA_USER_CONFIG_DIR}/.environment|" "${INSTALLER_DIR}/packaging/installer/netdata-updater.sh" > "${NETDATA_PREFIX}/usr/libexec/netdata-updater" || exit 1
```
Comment (cakrit) [inline netdata-installer.sh]: /usr/libexec/netdata-updater.sh

File: netdata-installer.sh [hunk 2]
```diff
@@ -958,39 +958,58 @@ To start netdata run:
...
+else
+	sed "s|THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT|${NETDATA_USER_CONFIG_DIR}/.environment|" "${NETDATA_SOURCE_DIR}/packaging/installer/netdata-updater.sh" > "${NETDATA_PREFIX}/usr/libexec/netdata-updater" || exit 1
```
Comment (cakrit) [inline netdata-installer.sh]: /usr/libexec/netdata-updater.sh

File: netdata-installer.sh [hunk 3]
```diff
@@ -958,39 +958,58 @@ To start netdata run:
...
+fi
 
-			rm -f "${crondir}/netdata-updater"
-			if [ -f "${INSTALLER_DIR}/packaging/installer/netdata-updater.sh" ]; then
-				sed "s|THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT|${NETDATA_USER_CONFIG_DIR}/.environment|" "${INSTALLER_DIR}/packaging/installer/netdata-updater.sh" > "${crondir}/netdata-updater" || exit 1
-			else
-				sed "s|THIS_SHOULD_BE_REPLACED_BY_INSTALLER_SCRIPT|${NETDATA_USER_CONFIG_DIR}/.environment|" "${NETDATA_SOURCE_DIR}/packaging/installer/netdata-updater.sh" > "${crondir}/netdata-updater" || exit 1
-			fi
+chmod 0755 ${NETDATA_PREFIX}/usr/libexec/netdata-updater
+echo >&2 "Update script is located at ${TPUT_GREEN}${TPUT_BOLD}${NETDATA_PREFIX}/usr/libexec/netdata-updater${TPUT_RESET}"
```
Comment (cakrit) [inline netdata-installer.sh]: /usr/libexec/netdata-updater.sh

File: netdata-installer.sh [hunk 4]
```diff
@@ -958,39 +958,58 @@ To start netdata run:
...
+	if [ "${AUTOUPDATE}" = "1" ]; then
+		if [ -f "${crondir}/netdata-updater.sh" ]; then
+			progress "Removing incorrect netdata-updater filename in cron"
+			rm -f "${crondir}/netdata-updater.sh"
+		fi
+
+		echo >&2 "Adding to cron"
+
+		rm -f "${crondir}/netdata-updater"
+		ln -sf "${NETDATA_PREFIX}/usr/libexec/netdata-updater" "${crondir}/netdata-updater"
```
Comment (cakrit) [inline netdata-installer.sh]: /usr/libexec/netdata-updater.sh

File: netdata-installer.sh [hunk 5]
```diff
@@ -958,39 +958,58 @@ To start netdata run:
...
+
+		echo >&2 "Auto-updating has been enabled. Updater script linked to: ${TPUT_RED}${TPUT_BOLD}${crondir}/netdata-update${TPUT_RESET}"
```
Comment (cakrit) [inline netdata-installer.sh]: netdata-updater

Comment (paulkatsoulakis) [reply netdata-installer.sh]: Good point, adjusting

Comment (cakrit) [reply netdata-installer.sh]: I still see netdata-update in the message

**Pull request merged**